src/api/app/models/staging/staged_requests.rb

```diff
@@ -84,6 +85,46 @@ def requests
     staging_workflow.unassigned_requests.where(number: request_numbers)
   end
 
+  def link_package(bs_request_action)
+    request = bs_request_action.bs_request
+
+    source_package = Package.get_by_project_and_name!(bs_request_action.source_project,
+                                                      bs_request_action.source_package)
+
+    # we need the revision and md5 and make it sticky in the _link
+    # packages in staging a locked in a specific version
+    query_options = { expand: 1 }
+    query_options[:rev] = bs_request_action.source_rev if bs_request_action.source_rev
+
+    backend_package_information = source_package.dir_hash(query_options)
+
+    # get the revision from the source package
+    source_rev = backend_package_information['vrev']
```

Comment:
Whatever the vrev is (I don't think there is any documentation about it :), it's not the source revision :)

Reply:
done

Reply:
vrev is the "version" revision, it is also incremented as rev, but it gets reset if the version of the package changes

Reply:
done